src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -700,6 +700,10 @@ public bool EnableBroadcast
         {
             get
             {
+                if (SocketType != SocketType.Dgram)
```

Comment:
```suggestion
                if (SocketType == SocketType.Stream)
```
Looks like `SO_BROADCAST` [also works for `SOCK_RAW`](https://docs.microsoft.com/en-us/windows/win32/winsock/tcp-ip-raw-sockets-2#send-and-receive-operations). Better be on the safe side and skip the OS PAL only for stream sockets.